test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -102,92 +98,119 @@
       (recur (inc i)
              (queue/cons-attempt result (Exception. (str "thud-" i)))))))
 
+(defn task-count [delay-pool]
+  (-> delay-pool
+      :pool-atom
+      deref
+      :thread-pool
+      .getQueue
+      count))
+
+(defn store-command' [q old-command]
+  (apply tqueue/store-command q (unroll-old-command old-command)))
+
+(def default-timeout-in-ms
+  (* 1000 60 5))
+
+(defn take-with-timeout!!
+  "Takes from `port` via <!!, but will throw an exception if
+  `timeout-in-ms` expires"
+  [port timeout-in-ms]
+  (async/alt!!
+    (async/timeout timeout-in-ms)
+    (throw (Exception. (format "Take from channel failed after timeout of '%s' ms was exceeded" timeout-in-ms)))
```

Comment:
Code line's a bit long, maybe wrap it a bit, and could perhaps get away with just "Channel take timed out after '%s' ms" or something.